### PR TITLE
Plane: Loiter_to_alt sing/climb rate override

### PR DESCRIPTION
**Pull request by magicrub**
Plane: Loiter_to_alt sing/climb rate override. mission item param 3 to dictate both the sink and climb rate while executing this mission item. If < 0.1 then the TECS param is used. This implementation ties climb/sink to same value.

Closes https://github.com/ArduPilot/ardupilot/issues/10487

### CODE REVIEW

**Comment by custeve** [comment]
@magicrub - I am able to test this in both SITL and eventually on live aircraft. I'm inexperienced at testing branches, so I may need to get up to speed on that.

File: libraries/AP_Mission/AP_Mission.h [hunk 1]
```diff
@@ -184,6 +184,12 @@ class AP_Mission {
         uint8_t relative_angle; // 0 = absolute angle, 1 = relative angle
     };
 
+    // LOITER_TO_ALT structure
+    struct PACKED Loiter_To_Alt_Command {
```
Comment (WickedShell) [inline libraries/AP_Mission/AP_Mission.h]: This is to large to fit in storage. How are you handling this? You only get 2 bytes past the location struct, anything else is corrupted/dropped.

Comment (magicrub) [reply libraries/AP_Mission/AP_Mission.h]: I'm not handling it. It's broken. I'm a dumbass

Comment (magicrub) [reply libraries/AP_Mission/AP_Mission.h]: This issue https://github.com/ArduPilot/ardupilot/issues/2391 is looooong overdue

**Comment by magicrub** [comment]
@custeve sorry, don't bother testing it. @WickedShell reminded me that I was an idiot and forgot that our mission item storage handing is maxed out. If we're storing a location (lat, lng, alt, flags) then we only have space for 16bits and that's currently helds with the radius.

**Comment by custeve** [comment]
@magicrub thanks for the update, I'm sure there's a solution. I'll standby.

File: libraries/AP_Mission/AP_Mission.cpp [hunk 2]
```diff
@@ -771,9 +771,10 @@ MAV_MISSION_RESULT AP_Mission::mavlink_int_to_mission_cmd(const mavlink_mission_
         break;
 
     case MAV_CMD_NAV_LOITER_TO_ALT:                     // MAV ID: 31
-        cmd.p1 = fabsf(packet.param2);                  // param2 is radius in meters
+        cmd.content.loiter_to_alt.radius = fabsf(packet.param2);    // radius in meters
         cmd.content.location.loiter_ccw = (packet.param2 < 0);
```
Comment (OXINARF) [inline libraries/AP_Mission/AP_Mission.cpp]: You can't set two different structs inside an union, one will overwrite the other.

Comment (magicrub) [reply libraries/AP_Mission/AP_Mission.cpp]: Yup. I know. Totally forgot. My bad.

**Comment by custeve** [comment]
@magicrub - I know this PR is closed, so pardon the question. Can you get around the 16 bit limit by limiting struct to a 10  bit radius (meters / 10) (10 km) and a 6 bit rate (rate m/s * 10) . TECS doesn't allow a vertical rate over 5.0, and this would provide 10km radius at a 10m resolution: 

```
struct PACKED LOITER_TO_ALT_COMMAND{
uint16_t radius:10;        // abs(radius) packet.param2 in meters / 10
uint8_t rate:6;             // climb/sink rate in meters/second * 10. If positive, it temporarily overrides the system param for both sink and climb rate
}
```
I've done something like this in C before, but not sure if that's legal syntax in this situation. An alternate option would be to add a parameter in TECS for an override rate, then use 15 bits for the radius and 1 bit as boolean flag for the override rate. The mission would have to set the override rate parameter before calling the command.

**Pull request closed without merging**